sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleClient.scala

```diff
@@ -33,8 +33,27 @@ import org.apache.spark.storage.ShuffleBlockBatchId
   * expected number of batches (Tables) is, and the ids for each table as they are received.
   */
 trait RapidsShuffleFetchHandler {
+  /**
+    * After a response for shuffle metadata is received, the expected number of columnar
+    * batches is communicated back to the caller via this method.
+    * @param expectedBatches - number of columnar batches
+    */
   def start(expectedBatches: Int): Unit
+
+  /**
+    * Called when a buffer is received, and has been handed off to the catalog.
+    * @param bufferId - a tracked shuffle buffer id
+    */
   def batchReceived(bufferId: ShuffleReceivedBufferId): Unit
+
+  /**
+    * Called when an error is detected during a metadata or a buffer fetch.
+    *
+    * @note No retry mechanism currently exists before calling [[transferError]],
```

Comment:
This note seems like an implementation detail of the code using this interface and not so relevant to those implementing the interface.  Does this make sense to place here?  Will we remember to update the comment when that is no longer the case, given the code that will be updated isn't here?